networkit/cpp/graph/test/GraphGTest.cpp

```diff
@@ -2169,6 +2185,34 @@ TEST_P(GraphGTest, testCompactEdges) {
     });
 }
 
+TEST(GraphGTest, testCompactEdgesAllVariations) {
+    Graph G = Graph(5);
+
+    G.addEdge(0, 1);
+    G.addEdge(1, 3);
+    G.addEdge(2, 3);
+    G.addEdge(0, 4);
+    G.addEdge(0, 3);
+
+    Graph GW = Graph(G, true, false);
+    Graph D = Graph(G, false, true);
+    Graph DW = Graph(G, true, true);
+
+    G.removeEdge(0, 4);
+    GW.removeEdge(0, 4);
+    D.removeEdge(0, 4);
+    DW.removeEdge(0, 4);
+
+    G.compactEdges();
+    ASSERT_EQ(G.numberOfEdges(), 4);
+    GW.compactEdges();
+    ASSERT_EQ(GW.numberOfEdges(), 4);
+    D.compactEdges();
+    ASSERT_EQ(D.numberOfEdges(), 4);
+    DW.compactEdges();
+    ASSERT_EQ(DW.numberOfEdges(), 4);
```

Comment:
Here `EXPECT_EQ` would be more appropriate, https://testing.googleblog.com/2008/07/tott-expect-vs-assert.html